contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -477,13 +499,43 @@ abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
 
     function _approveBase() internal virtual {
         IERC20 pToken = IERC20(platformAddress);
+
         // Balancer vault for BPT token (required for removing liquidity)
         pToken.safeApprove(address(balancerVault), type(uint256).max);
     }
 
+    /**
+     * @notice Returns the rate supplied by the Balancer configured rate
+     * provider. Rate is used to normalize the token to common underlying
+     * pool denominator. (ETH for ETH Liquid staking derivatives)
+     *
+     * @param _asset Address of the Balancer pool asset
+     * @return rate of the corresponding asset
+     */
     function _getRateProviderRate(address _asset)
         internal
         view
         virtual
-        returns (uint256);
+        returns (uint256)
```

Comment:
Does a local cache work for this pool type?

Reply:
Yes thanks for reminding me. Cached in this commit: [8c8732004095f7c29ea7b569f6a45792f3dce62b](https://github.com/OriginProtocol/origin-dollar/pull/1839/commits/8c8732004095f7c29ea7b569f6a45792f3dce62b)